examples/gno.land/p/demo/simpledao/propstore.gno

```diff
@@ -0,0 +1,173 @@
+package simpledao
+
+import (
+	"errors"
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/seqid"
+	"gno.land/p/gov/dao"
+)
+
+var (
+	errMissingProposal = errors.New("proposal is missing")
+	errMissingVote     = errors.New("member has not voted")
+)
+
+const (
+	// maxRequestProposals is the maximum number of
+	// paginated proposals that can be requested
+	maxRequestProposals = 10
+
+	// maxRequestMembers is the maximum number of
+	// paginated members that can be requested
+	maxRequestMembers = 50
+)
+
+// proposal is the internal simpledao proposal implementation
+type proposal struct {
+	author      std.Address // initiator of the proposal
+	description string      // description of the proposal
+
+	executor dao.Executor       // executor for the proposal
+	status   dao.ProposalStatus // status of the proposal
+
+	votes *votes // voting mechanism
+}
+
+func (p *proposal) GetAuthor() std.Address {
+	return p.author
+}
+
+func (p *proposal) GetDescription() string {
+	return p.description
+}
+
+func (p *proposal) GetStatus() dao.ProposalStatus {
+	return p.status
+}
+
+func (p *proposal) GetVotes() []dao.Vote {
```

Comment:
I do not understand why we need an avl.Tree to store all the votes, but we have a method that converts it to a slice...

Reply:
We use the `avl.Tree` as a map to easily track votes (address -> vote option)

Reply:
I still think a slice will be much easier and faster below 1k elements, but it is up to you, I still don't have benchmarks to demonstrate my assumptions about it.